canonical-web-and-design/maas-ui
# Add react error reporting with sentry js.

## Done
* Added sentry DSN to .env.production.
* Capture exceptions for sentry in ErrorBoundary, including maas version.
* Send sentry error reports within react if enableAnalytics config enabled.
* Update the `enable_analytics` checkbox copy to be more generic.

## Screenshot
![image](https://user-images.githubusercontent.com/130286/73117656-d6fc9600-3f51-11ea-9243-77bb9489af21.png)

## QA
* Configure a sentry DSN for development:
add `REACT_APP_SENTRY_DSN=https://838f6da6d45b4513bae7398d0327d2a7@sentry.is.canonical.com//21` to `.env.development` and restart webpack.
* Create some broken code in the react app (e.g. `throw new Error('error')`).
* Trigger the error. You should see it reported against the [Test project](https://sentry.is.canonical.com/canonical/test/) on sentry.
* Confirm that the stacktrace and maas version have been set correctly, e.g.
![image](https://user-images.githubusercontent.com/130286/73318156-6df08780-429d-11ea-935f-564d7367cc72.png)

Fixes #620
